CodeAnalyzerTestProject/Assets/Tests/CleanConsoleTest.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using NUnit.Framework;
+using UnityEditor.PackageManager;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Is = UnityEngine.TestTools.Constraints.Is;
+
+
+namespace PackageTestSuite
+{
+    public class CleanConsoleTest
+    {
+        // Make sure it is run before anything else that can affect the console
+        [Test, Order(1)]
+        public void TestCleanConsole()
+        {
+            string logs = Utilities.GetLogs();
+			Assert.That(logs, Is.Empty, string.Format("Found logs in the console on project startup.\n\nLogs: {0}",logs));

```

Comment:
Isn't this one already performed automatically by the package checks?

Reply:
Yes (this test is copied from the package validation suite), _but_ in this project we enabled the code analyzer (using the csc.rsp file), so we will also get warnings from the code analyzer this way (and, in case you want to ask next: yes, it would be nicer if we would not copy the code, but use it from the suite, but no, doing that is not at all easy).